pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -185,11 +193,16 @@ func (c *controller) List(typ, namespace string) ([]model.Config, error) {
 
 		switch typ {
 		case model.VirtualService.Type:
-			_, virtualServices := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
-			out = append(out, virtualServices)
+			ConvertIngressVirtualService(*ingress, c.domainSuffix, ingressByHost)
 		case model.Gateway.Type:
-			gateways, _ := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
-			out = append(out, gateways)
+			//gateways := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
+			//out = append(out, gateways)
+		}
```

Comment:
The auto generated gateway was a big deal for the ingress. As such our conversion tool is sufficient to convert to gateway and virtual service.

Reply:
I can uncomment it - but it doesn't really work. Secrets from TLS can't be loaded (there is no code to read the secrets in ingress - they need to be mounted). Other than that - a host-* and port 80/443 is all gateway needed, why generate one gateway per ingress ? 

I will add the Gateway with the proper name to helm - so users will not have to do anything, but separate PR since there is a whole separate discussion on how to setup secrets for TLS and which LB to use.